File: frontend/store/Expense.js
import { observable } from "mobx"

export default class Expense {
    @observable expenseSheetName
    @observable id
    @observable date
    @observable description
    @observable category
    @observable paidBy
    @observable amount

    constructor(expenseSheetName, id, dateStr, description, category, paidBy, amount) {
        this.expenseSheetName = expenseSheetName
        if (!id) {
            id = -1
        }
        if(!dateStr) {
            this.date = new Date()
        } else {
            this.date = new Date(Date.parse(dateStr))
        }
        this.id = id
        this.description = description
        this.category = category
        this.paidBy = paidBy
        this.amount = amount
    }
}

File: backend/src/com/mehtasankets/mmflow/domain/Expense.kt
package com.mehtasankets.mmflow.domain

import java.time.Instant

data class Expense (
    var expenseSheetName: String = "",
    var id: Long = 0,
    var date: Instant = Instant.now(),
    var description: String = "",
    var category: String = "",
    var paidBy: String = "",
    var amount: Double = 0.0
)